internal/cmd/gocdk/main.go

```diff
@@ -127,3 +132,29 @@ func findModuleRoot(ctx context.Context, dir string) (string, error) {
 	}
 	return string(output), nil
 }
+
+// withInterrupt returns a copy of parent with a new Done channel. The returned
+// context's Done channel will be closed when the process receives an interrupt
+// signal, the parent context's Done channel is closed, or the stop function is
+// called, whichever comes first.
+//
+// The stop function releases resources and stops listening for signals, so code
+// should call it as soon as the operation using the context completes.
+func withInterrupt(parent context.Context) (_ context.Context, stop func()) {
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, interruptSignals()...)
+	ctx, cancel := context.WithCancel(parent)
+	done := make(chan struct{})
+	go func() {
+		select {
+		case <-sig:
+			cancel()
+		case <-done:
```

Comment:
Does this goroutine ever exit if the parent context is cancelled? Maybe add `<-ctx.Done()`?